pkg/controller/trainjob_controller.go

```diff
@@ -105,11 +107,29 @@ func (r *TrainJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 func (r *TrainJobReconciler) reconcileObjects(ctx context.Context, runtime jobruntimes.Runtime, trainJob *trainer.TrainJob) (objsOpState, error) {
 	log := ctrl.LoggerFrom(ctx)
 
-	objs, err := runtime.NewObjects(ctx, trainJob)
+	objects, err := runtime.NewObjects(ctx, trainJob)
 	if err != nil {
 		return buildFailed, err
 	}
-	for _, obj := range objs {
+	for _, object := range objects {
+		// TODO (astefanutti): Remove conversion to unstructured when the runtime.ApplyConfiguration
+		//  interface becomes available and first-class SSA method is added to the controller-runtime
+		// client. See https://github.com/kubernetes/kubernetes/pull/129313
+		var obj client.Object
+		if o, ok := object.(client.Object); ok {
```

Comment:
In this case, could we just return build error since this case has potential unintended null defaulting?
And our in-tree plugins never see this case, this is only for external-tree plugins.

Reply:
You're right, currently the in-tree plugins never see this case.